Samples/SendReceiveBatch/Receiver.cs

```diff
@@ -0,0 +1,66 @@
+using AzureBatchQueue;
+using Microsoft.Extensions.Logging;
+
+namespace SendReceiveBatch;
+
+public class Receiver
+{
+    private BatchQueue<string> batchQueue;
+    public string[] DoNotComplete = { "four", "six" };
+    private bool canceled = false;
+
+    public Receiver(string queueName, ILogger<BatchQueue<string>>? logger)
+    {
+        batchQueue = new BatchQueue<string>("UseDevelopmentStorage=true", queueName, flushPeriod: TimeSpan.FromSeconds(5), logger: logger);
+    }
+
+    public async Task Init()
+    {
+        await batchQueue.Init();
+        Log("Receiver is ready.");
+    }
+
+    public async Task Receive()
+    {
+        Log("Start receiving.");
+
+        while (!canceled)
+        {
+            var batchItems = await batchQueue.ReceiveBatch();
+
+            Log($"Received batch with {batchItems.Length} items.");
+
+            foreach (var batchItem in batchItems)
+            {
+                if (DoNotComplete.Contains(batchItem.Item))
+                {
+                    Log($"Will not complete {batchItem.Item}");
+                    continue;
+                }
+
+                Log($"Completed {batchItem.Item}");
+                await batchItem.Complete();
```

Comment:
Do we need `async` here? According to the code, it's not needed here.

Conceptually, I see two options:
1. `async` and block on completing the last item in a batch.
   Pros: We have backpressure from the queue. E.g.,  if it's busy or unavailable, the client code will be aware of that.
   Cons: A little bit cumbersome async API for the client. We actually need to wait only once in a while (when completing the last item in a batch).
     
2. Don't block on complete. Batch completion always happens in a background timer thread.

Backpressure is usually a good thing in distributed systems. But the simplicity of sync API is tempting, too. My gut feeling says the first approach is right, but I'd still try to test the second one under load.

Reply:
I decided to go with option 2 now and will test it under load.